### PR TITLE
Add a warning when not running in optimized mode

### PR DESCRIPTION
**Pull request by davfsa**
### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: hikari/internal/ux.py [hunk 1]
```diff
@@ -233,6 +240,16 @@ def print_banner(
         stdout.write(string.Template(raw_banner).safe_substitute(args))
 
 
+def warn_if_not_optimized() -> None:
+    """Log a warning if not running in optimization mode."""
+    if __debug__:
+        _LOGGER.warning(
+            "You are running on no-optimization level, which may speed-down your application. "
```
Comment (GoogleGenius) [inline hikari/internal/ux.py:247]:
```suggestion
            "You are not running on any optimization level, which may slow down your application. "
```

Comment (davfsa) [reply hikari/internal/ux.py:247]: Would this be better?

```py
        "You are running on optimization level 0 (no optimizations), which may slow down your application. "
```

as it's not that you are not running on an optimization level, you are just running on optimization level 0.

Comment (GoogleGenius) [reply hikari/internal/ux.py:247]: I think that works!

Comment (davfsa) [reply hikari/internal/ux.py:247]: Cool, I'll change it when I have a computer to ensure it doesn't go over the character limit.

Thanks for the suggestion!

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**